File: app/src/main/java/com/satsumasoftware/pokedex/framework/nature/MiniNature.java
package com.satsumasoftware.pokedex.framework.nature;

import android.content.Context;
import android.database.Cursor;
import android.database.sqlite.SQLiteDatabase;
import android.os.Parcel;
import android.os.Parcelable;

import com.satsumasoftware.pokedex.db.NaturesDBHelper;

public class MiniNature extends BaseNature implements Parcelable {

    public MiniNature(int id, String name) {
        mId = id;
        mName = name;
    }

    public Nature toNature(Context context) {
        NaturesDBHelper helper = new NaturesDBHelper(context);
        SQLiteDatabase db = helper.getReadableDatabase();
        Cursor cursor = db.query(
                NaturesDBHelper.TABLE_NAME,
                null,
                NaturesDBHelper.COL_ID + "=?",
                new String[] {String.valueOf(mId)},
                null,
                null,
                null);
        cursor.moveToFirst();
        Nature nature = new Nature(cursor);
        cursor.close();
        return nature;
    }


    protected MiniNature(Parcel in) {
        mId = in.readInt();
        mName = in.readString();
    }

    @Override
    public void writeToParcel(Parcel dest, int flags) {
        dest.writeInt(mId);
        dest.writeString(mName);
    }

    @Override
    public int describeContents() {
        return 0;
    }

    public static final Creator<MiniNature> CREATOR = new Creator<MiniNature>() {
        @Override
        public MiniNature createFromParcel(Parcel in) {
            return new MiniNature(in);
        }

        @Override
        public MiniNature[] newArray(int size) {
            return new MiniNature[size];
        }
    };
}

File: app/src/main/java/com/satsumasoftware/pokedex/framework/nature/MiniNature.kt
package com.satsumasoftware.pokedex.framework.nature

import android.content.Context
import android.os.Parcel
import android.os.Parcelable
import com.satsumasoftware.pokedex.db.NaturesDBHelper

class MiniNature(val id: Int, val name: String) : Parcelable {

    fun toNature(context: Context): Nature {
        val helper = NaturesDBHelper.getInstance(context)
        val cursor = helper.readableDatabase.query(
                NaturesDBHelper.TABLE_NAME,
                null,
                NaturesDBHelper.COL_ID + "=?",
                arrayOf(id.toString()),
                null,
                null,
                null)
        cursor.moveToFirst()
        val nature = Nature(cursor)
        cursor.close()
        return nature
    }

    constructor(source: Parcel) : this(source.readInt(), source.readString())

    override fun describeContents() = 0

    override fun writeToParcel(dest: Parcel?, flags: Int) {
        dest?.writeInt(id)
        dest?.writeString(name)
    }

    companion object {

        @JvmField val CREATOR: Parcelable.Creator<MiniNature> = object : Parcelable.Creator<MiniNature> {
            override fun createFromParcel(source: Parcel): MiniNature = MiniNature(source)
            override fun newArray(size: Int): Array<MiniNature?> = arrayOfNulls(size)
        }

        @JvmField val DB_COLUMNS = arrayOf(NaturesDBHelper.COL_ID, NaturesDBHelper.COL_NAME)

    }
}
